pynucastro/networks/rate_collection.py

```diff
@@ -547,6 +598,92 @@ def plot(self, trace_threshold=0.1, hard_limit=None, size=(9, 5)):
         return fig
 
 
+class LoddersComposition(Composition):
+    """A class to use present day solar abundances from Lodders et al. 2020 & 2021.
+
+    Parameters
+    ----------
+    z : float
+        Target metallicity :math:`Z` to scale the Lodders solar
+        mixture to. If ``None`` (the default), the unscaled Lodders
+        abundances are used.
+    half_life_thresh : float
+        The half life value below which to zero the mass fraction
+        of a nucleus.  This prevents us from making a composition
+        that is not really stable.
+
+    Notes
+    -----
+    The Lodders abundances are stored in ``LODDERS_DATA`` as mass
+    fractions for individual isotopes.
+    """
+
+    def __init__(self, z=None, half_life_thresh=None):
+        nuclei = []
+        for key in LODDERS_DATA:
+            if key == "h1":
+                name = "p"
+            else:
+                name = key
+            nuclei.append(Nucleus(name))
+
+        # base composition initialize in Composition
+        super().__init__(nuclei)
+
+        # now give lodders abundances with scaling if needed
+
+        self._get_from_lodders(z, half_life_thresh=half_life_thresh)
+
+    def _get_from_lodders(self, z=None, half_life_thresh=None):
+
+        x_solar = 0.0
+        y_solar = 0.0
+
+        for key, X_i in LODDERS_DATA.items():
```

Comment:
the mass fractions from `LODDERS_DATA` sum to `1.00004155`, so we should do a renormalization before we actually use them.